src/Mod/Path/Path/Base/Generator/helix.py

```diff
@@ -229,16 +238,18 @@ def retract():
         # safe place which does not touch the inner or outer wall on all radii except
         # the first.  This is left as a future improvement.
 
-        retractcommands.append(Path.Command("G0", {"Z": startPoint.z}))
+        retractcommands.append(Path.Command("G0", {"Z": clearanceHeight if isLast else safeHeight}))
```

Comment:
FYI: IIRC the final retract to `ClearanceHeight` is added by the framework automatically after each operation.

Reply:
The operation may consist of **multiple** holes. I chose to retract to `ClearanceHeight` after each hole, before the G0 to the next hole.
Indeed there is an extra retract after the last hole. That does not cause any problem.